embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/logs/EmbraceLogService.kt

```diff
@@ -0,0 +1,128 @@
+import io.embrace.android.embracesdk.worker.BackgroundWorker
+import io.opentelemetry.api.common.AttributeKey
+import io.opentelemetry.api.logs.Logger
+import java.util.concurrent.TimeUnit
+
+/**
+ * Creates log records to be sent using the Open Telemetry Logs data model.
+ */
+internal class EmbraceLogService(
+    private val logger: Logger,
+    private val metadataService: MetadataService,
+    private val sessionIdTracker: SessionIdTracker,
+    private val backgroundWorker: BackgroundWorker
+) : LogService {
+
+    override fun log(
+        message: String,
+        severity: Severity,
+        properties: Map<String, Any>?
+    ) {
+        val sessionId = sessionIdTracker.getActiveSessionId()
+        val appState = metadataService.getAppState()
+
+        backgroundWorker.submit {
+            // TBD: Check if config enabled
+            // TBD: Check if log should be gated
+            // TBD: Count log and enforce limits
+
+            emitLog(
+                message,
+                severity,
+                sessionId,
+                appState,
+                properties
+            )
+        }
+    }
+
+    override fun logException(
+        message: String,
+        logExceptionType: LogExceptionType,
+        properties: Map<String, Any>?,
+        stackTraceElements: Array<StackTraceElement>?,
+        customStackTrace: String?,
+        context: String?,
+        library: String?,
+        exceptionName: String?,
+        exceptionMessage: String?
+    ) {
+        val sessionId = sessionIdTracker.getActiveSessionId()
+        val appState = metadataService.getAppState()
+
+        backgroundWorker.submit {
+            // TBD: Check if config enabled
+            // TBD: Check if log should be gated
+            // TBD: Count log and enforce limits
+
+            val attributes = mutableMapOf<String, Any>()
+            properties?.let { attributes.putAll(it) }
+
+            emitLog(
+                message,
+                Severity.ERROR,
+                sessionId,
+                appState,
+                properties,
+                logExceptionType,
+                exceptionName,
+                exceptionMessage,
+                context,
+                library
+            )
+        }
+    }
+
+    private fun emitLog(
+        message: String,
+        severity: Severity,
+        sessionId: String?,
+        appState: String?,
+        properties: Map<String, Any>?,
+        exceptionType: LogExceptionType? = null,
+        exceptionName: String? = null,
+        exceptionMessage: String? = null,
+        context: String? = null,
+        library: String? = null
+    ) {
+        val otelSeverity = mapSeverity(severity)
+
+        val builder = logger.logRecordBuilder()
+            .setBody(message)
+            .setSeverity(otelSeverity)
+            .setSeverityText(otelSeverity.name)
+            .setTimestamp(System.currentTimeMillis(), TimeUnit.MILLISECONDS)
```

Comment:
```suggestion
            .setTimestamp(clock.now(), TimeUnit.MILLISECONDS)
```

We probably want to use the `Clock` implementation here for testability